src/navigation_controller.ts

```diff
@@ -557,22 +557,40 @@ export class NavigationController {
           workspace.keyboardAccessibilityMode &&
           !workspace.options.readOnly
         ) {
-          const curNode = workspace.getCursor()?.getCurNode();
-          if (curNode && curNode.getSourceBlock()) {
-            const sourceBlock = curNode.getSourceBlock();
-            return !!(
-              !Blockly.Gesture.inProgress() &&
-              sourceBlock &&
-              sourceBlock.isDeletable() &&
-              sourceBlock.isMovable()
-            );
+          switch (this.navigation.getState(workspace)) {
+            case Constants.STATE.WORKSPACE:
+              const curNode = workspace?.getCursor()?.getCurNode();
+              if (curNode && curNode.getSourceBlock()) {
+                const sourceBlock = curNode.getSourceBlock();
+                return !!(
+                  !Blockly.Gesture.inProgress() &&
+                  sourceBlock &&
+                  sourceBlock.isDeletable() &&
+                  sourceBlock.isMovable()
+                );
+              }
+              return false;
```

Comment:
```suggestion
              const sourceBlock = curNode?.getSourceBlock();
              return !!(
                sourceBlock?.isDeletable() &&
                sourceBlock?.isMovable() &&
                !Blockly.Gesture.inProgress()
              );
```
`sourceBlock` might be null/undefined but I _think_ the final result will be the same in all cases.  I reordered the clauses in the final expression to match the order in the following case (but maybe it would be better to reorder that one instead).

Reply:
Applied manually.